topostats/tracing/dnatracing.py

```diff
@@ -744,20 +714,31 @@ def trace_image(
         )
         LOGGER.info(f"[{filename}] : Traced grain {n_grain + 1} of {n_grains}")
         ordered_traces.append(result.pop("ordered_trace"))
+        splined_traces.append(result.pop("splined_trace"))
         results[n_grain] = result
         n_grain += 1
     try:
         results = pd.DataFrame.from_dict(results, orient="index")
         results.index.name = "molecule_number"
         image_trace = trace_mask(grain_anchors, ordered_traces, image.shape, pad_width)
+        # rounded_splined_traces = [np.round(np.array(splined_trace)).astype(int) for splined_trace in splined_traces]
+        rounded_splined_traces = []
+        for splined_trace in splined_traces:
+            if splined_trace is not None:
+                rounded_splined_trace = np.round(splined_trace).astype(int)
+                rounded_splined_traces.append(rounded_splined_trace)
+        rounded_splined_traces = np.array(rounded_splined_traces)
+        image_spline_trace = trace_mask(grain_anchors, rounded_splined_traces, image.shape, pad_width)
     except ValueError as error:
         LOGGER.error("No grains found in any images, consider adjusting your thresholds.")
         LOGGER.error(error)
     return {
         "statistics": results,
         "ordered_traces": ordered_traces,
+        "splined_traces": splined_traces,
         "cropped_images": cropped_images,
         "image_trace": image_trace,
```

Comment:
Hi, I had a quick look at the code, and I noticed that you removed line 428 in processing.py, `image_trace = tracing_results["image_trace"]`, but you kept the `image_trace` here. Is this intentional, and is this `image_trace` supposed to be different from both the ordered and the splined trace?

Reply:
Yeah this is intentional as I foresee a need to have `image_trace` later on for image output, or for development purposes but we don't currently need to plot it. It is also useful for people using notebooks to have access to.